external/leveldb/leveldb_tuning.ipynb

```diff
@@ -0,0 +1,767 @@
+    "initialize_optimizer()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 46,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [
+    {
+     "output_type": "stream",
+     "name": "stdout",
+     "text": [
+      "Parameter value: 14231436, Objective value: 21.8 MB/s\n",
+      "Parameter value: 7465366, Objective value: 21.1 MB/s\n",
+      "Optimal parameter: 14231436, Optimal value: 21.8\n",
+      "Parameter value: 61708892, Objective value: 21.9 MB/s\n",
+      "Optimal parameter: 14231436, Optimal value: 21.8\n",
+      "Parameter value: 13801008, Objective value: 21.5 MB/s\n",
+      "Optimal parameter: 61708892, Optimal value: 21.9\n",
+      "Parameter value: 37672330, Objective value: 21.3 MB/s\n",
+      "Optimal parameter: 61708892, Optimal value: 21.9\n",
+      "Parameter value: 56044662, Objective value: 21.7 MB/s\n",
+      "Optimal parameter: 61708892, Optimal value: 21.9\n",
+      "Parameter value: 4266076, Objective value: 20.8 MB/s\n",
+      "Optimal parameter: 61708892, Optimal value: 21.9\n",
+      "Parameter value: 25616874, Objective value: 22.4 MB/s\n",
+      "Optimal parameter: 61708892, Optimal value: 21.9\n",
+      "Parameter value: 55095476, Objective value: 20.5 MB/s\n",
+      "Optimal parameter: 25616874, Optimal value: 22.4\n",
+      "Parameter value: 2542654, Objective value: 22.0 MB/s\n",
+      "Optimal parameter: 25616874, Optimal value: 22.4\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Please change the leveldb_path to the build directory of your leveldb installation\n",
+    "leveldb_path = \"$HOME/leveldb/build/\"\n",
+    "# You can change the command to run a different kind of workload (take a look at db_bench.cc to see the possible workloads)\n",
+    "command = \"db_bench\"\n",
+    "\n",
+    "# You might have to change the run workload function to explore a combination of parameters simultaneously\n",
+    "def run_workload(workload, input_parameter, parameter_value):\n",
+    "    # The line below executes the db_bench command with approprite parameters, you can change this \n",
+    "    # if you want to specify other input parameters\n",
+    "    result = subprocess.check_output(leveldb_path + command + \" --benchmarks=\" + workload +  \" --\" + str(input_parameter) + \"=\" + str(parameter_value), shell=True)\n",
+    "    stats = (str(result).split(\":\")[-1]).split(\";\")\n",
+    "    # The line below is used to parse the output that is returned by db_bench\n",
+    "    latency, throughput = float(stats[0].strip().split(\" \")[0]), float(stats[1].strip().split(\" \")[0])\n",
+    "    return latency, throughput\n",
+    "\n",
+    "#optimizer = bayesian_optimizer_factory.create_remote_optimizer(optimization_problem=optimization_problem)\n",
+    "def run_optimizer():\n",
+    "    # Parameter 1: write_buffer_size: min_value = 1 MB, max_value = 128 MB\n",
+    "    # Parameter 2: max_file_size: min_value = 1 MB, max_value = 128 MB\n",
```

Comment:
@nithinv13 as mentioned: although these might be the experimentally chosen values you used, we could consider to expand the range to ensure that a different machine/workload doesn't find something else.